server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -90,15 +95,26 @@ private static KeywordFieldMapper toType(FieldMapper in) {
 
         private final Parameter<Map<String, String>> meta = Parameter.metaParam();
 
+        private final Parameter<Script> script = Parameter.scriptParam(m -> toType(m).script);
+        private final Parameter<String> onScriptError = Parameter.onScriptErrorParam(m -> toType(m).onScriptError, script);
+
         private final IndexAnalyzers indexAnalyzers;
+        private final ScriptCompiler scriptCompiler;
 
-        public Builder(String name, IndexAnalyzers indexAnalyzers) {
+        public Builder(String name, IndexAnalyzers indexAnalyzers, ScriptCompiler scriptCompiler) {
             super(name);
             this.indexAnalyzers = indexAnalyzers;
+            this.scriptCompiler = scriptCompiler;
+            this.script.precludesParameters(nullValue);
```

Comment:
how about ignore_above and normalizer? Should they be supported when a script is set?

Reply:
Yes, I've included those - they're both things that are applied after a value has been read, as opposed to `null_value` which is applied *instead* of a value.